Given to user possibility to send a setting PlainObject into load function.

roger-russel opened the pull request:
It was necessary into one of my projects at work use an especific $.ajax setting.

Because this I implemented into Load function a handle for a PlainObject setting of $.ajax.

If it was some thing that I should change or correct please tell me.

Commit:
include netbeans project folder to ignore

Commit:
Handle an ajax Setting PlainObject passed by user on place of url string

include netbeans IDE trash to ignore

Test of a call with Setting PlainObject

it will use dataType on settings PlainObject by user, or if user don't set anyone will be used default dataType.

Throw an error if url is null.

corrected indexing.

Netbeans autoindex all file withou my concern.

Correct index and thow

Corrected the index, I really sorry my netbans was settup to index by it owns, so I correct directy on github page by now.
And I changed to Throw a new Error as you ask.

I hope everything is right in the code now, it it was I will make the squash like you ask.

Commit:
Handle an ajax Setting PlainObject passed by user on place of url string

Commit:
Test of a call with Setting PlainObject

jonnyreeves commented on src/jquery.mustache.js:
```diff
@@ -127,17 +127,23 @@
+
+		var obj = {};
+
+		obj = typeof url === 'object' ? obj = url: obj.url = url;
```
Do we need a null check here as well to avoid a NPE? (`typeof url === 'object' && !!url`)

roger-russel replied:
If the url is null will break anyway, what do you think of throw an exception?
Something Like this:

```
if( typeof url === 'undefined' ) 
 throw 'Url is needed';
```

jonnyreeves replied:
Yep, sounds reasonable, although we probably want to check for non-falsy, (ie: `if (!!url) throw`)

roger-russel replied:
Done, if url is null it will throw an exeception.

jonnyreeves replied:
Thanks!  Could you fix up the white space on your last commit and throw an
Error instance instead of a string, please.  Also, could you squash your
commits into one and I'll merge it through tonight.

Thanks again
Jonny.

On 21 July 2015 at 17:19, Roger Russel <notifications@github.com> wrote:

> In src/jquery.mustache.js
> <https://github.com/jonnyreeves/jquery-Mustache/pull/38#discussion_r35121956>
> :
>
> >  	 * @param onComplete	Optional callback function which will be invoked when the templates from the supplied URL
> >  	 *						have been loaded and are ready for use.
> >  	 * @returns				jQuery deferred promise which will complete when the templates have been loaded and are
> >  	 *						ready for use.
> >  	 */
> >  	function load(url, onComplete) {
> > -		return $.ajax({
> > -				url: url,
> > -				dataType: options.externalTemplateDataType
> > -			}).done(function (templates) {
> > +
> > +		var obj = {};
> > +
> > +		obj = typeof url === 'object' ? obj = url: obj.url = url;
>
> Done, if url is null it will throw an exeception.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jonnyreeves/jquery-Mustache/pull/38/files#r35121956>.
>

jonnyreeves commented on src/jquery.mustache.js:
```diff
@@ -127,17 +127,23 @@
+
+		obj = typeof url === 'object' ? obj = url: obj.url = url;
+
+		obj.dataType = options.externalTemplateDataType;
```
We should leave it as is if set by the user, otherwise it's just confusing.

roger-russel replied:
Like this:

```
obj.dataType = obj.dataType || options.externalTemplateDataType;
```

If user was settup a dataType, it will be used and if dont default DataType will be used.

jonnyreeves commented:
Looks good, couple of minor action points if you would be so kind?  Thanks for adding a test :+1:

Commit:
it will use dataType on settings PlainObject by user, or if user don't set anyone will be used default dataType.

Commit:
Throw an error if url is null.

Commit:
corrected indexing.

Commit:
Netbeans autoindex all file withou my concern.

roger-russel committed:
Correct index and thow

Corrected the index, I really sorry my netbans was settup to index by it owns, so I correct directy on github page by now.
And I changed to Throw a new Error as you ask.

I hope everything is right in the code now, it it was I will make the squash like you ask.

roger-russel commented:
I made the changes that were requested, sorry for the coding, were pre configured in my editor.

Is there anything else that needs to change, or something I did not understand what was right to do?

Have you thought about putting a contributing page like [this page](https://github.com/Codeception/Codeception/blob/2.1/CONTRIBUTING.md) and a [Editorconfig](http://editorconfig.org/) file in the project, like this [.editorconfig](https://github.com/locaweb/locawebstyle/blob/master/.editorconfig) file?

jonnyreeves commented:
Hi @RogerRussel, this looks good to land - could you just squash into a single commit please (and consider removing the modification to .gitignore)

Thanks again for the contribution.

Commit:
move netbeans ide trash to place os IDE trashes

Commit:
Merge branch 'master' of github.com:RogerRussel/jquery-Mustache

roger-russel committed:
Move Netbeans IDE Junk to right place
